Feature/editorial intro and format

mprorock opened the pull request:
Many line number changes here.  Actual changes in this:
1) Reformat and reflow for consistent indentation in the html and line breaks
2) retitle to "Securing Verifiable Credentials using JOSE and COSE" with new short name "vc-jose-cose"
3) reorder sections so securing core data model comes first
4) adjust abstract, and add introductory text that refers to items we are actually doing in this spec


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/mesur-io/vc-jwt/pull/112.html" title="Last updated on Jun 29, 2023, 7:21 PM UTC (b9bcdb8)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/112/bb5954f...mesur-io:b9bcdb8.html" title="Last updated on Jun 29, 2023, 7:21 PM UTC (b9bcdb8)">Diff</a>

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      and with CBOR Object Signing and Encryption (COSE) [[RFC8152]].
+
+      This enables the Verifiable Credential data model
+      [[VC-DATA-MODEL]]to be easily utilized with existing approaches
+      to signing and encryption that are widely adopted.
```
```suggestion
      [[VC-DATA-MODEL]]to be implemented with standards
      for signing and encryption that are widely adopted.
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      to ensure the integrity, authenticity, and non-repudiation of
+      the information contained in a JSON object. JWS enables secure
+      communication and trust between parties by adding a
+      cryptographic signature to the JSON payload. This signature is
```
```suggestion
      cryptographic signature to a payload. This signature is
```

OR13 replied:
JWS can secure arbitrary content, JWT can only secure JSON.

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      communication and trust between parties by adding a
+      cryptographic signature to the JSON payload. This signature is
+      generated using a private key and can be verified using the
+      corresponding public key. JWS ensures the tamper-proof nature of
+      the data, protecting it from unauthorized modifications. The
```
Im not attached  to any of this language, just trying to capture the essence of what digital signatures do

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      the data, protecting it from unauthorized modifications. The
+      versatility and wide adoption of JWS make it an essential
+      component for verifying the authenticity and integrity of
+      JSON-based messages in various applications, including
```
```suggestion
      data in various applications, including
```

OR13 replied:
again, JWS secures arbitrary bytes, JWT only secures JSON

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      JSON-based messages in various applications, including
+      authentication, authorization, and secure data exchange. These
+      properties make JWS especially well suited to securing the
+      [[VC-DATA-MODEL]] and through this to provide a means of
```
```suggestion
      JSON-LD [[VC-DATA-MODEL]] and through this to provide a means of
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+    <p>
+      JSON Web Token (JWT) [[RFC7519]] is a widely-used means of
+      expressing
+      <a data-cite="VC-DATA-MODEL#claims">claims</a> to be transferred
```
```suggestion
      <a data-cite="VC-DATA-MODEL#claims">claims</a>, represented in JSON, to be transferred
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      <a data-cite="VC-DATA-MODEL#claims">claims</a> as a JSON object
+      that is contained in a JSON Web Signature (JWS) [[RFC7515]]
+      and/or JSON Web Encryption (JWE) [[?RFC7516]]. For this
+      specification, the use of JWE is out of scope.
```
```suggestion
      specification, the use of JWE is out of scope.
      <p class="issue">The working group is discussing how we might comment on 
      the use of encryption in <a href="https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-3.2">OpenID for Verifiable Presentations</a>
      </p> 
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      cryptographic options, allowing for a wide range of algorithms
+      to be used for signing and encryption.
 
+      COSE supports two main operations: signing and encryption. For
```
nit, maybe a second p tag.

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
 
+      COSE supports two main operations: signing and encryption. For
+      signing, COSE allows the creation of digital signatures over
+      CBOR data using various algorithms such as HMAC, ECDSA, or
+      EdDSA. These signatures provide assurance of data integrity and
```
```suggestion
      CBOR data using various algorithms such as HMAC, RSA, ECDSA, or
      EdDSA. These signatures provide assurance of data integrity and
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      EdDSA. These signatures provide assurance of data integrity and
+      authenticity. COSE also supports encryption, enabling the
+      confidentiality of CBOR data by encrypting it with symmetric or
+      asymmetric encryption algorithms. Our use case here, with
+      verifiable credentials is focused on signing, primarily through
+      use of COSE Sign1 operations.
```
```suggestion
      asymmetric encryption algorithms.
```

OR13 replied:
Its true that COSE Sign1 is the best way to use COSE to secure vc+ld+json content type, but this seems like not the best place to say this.

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+  </section>
 
+  <section>
+    <h2>Securing the Core VC Data Model</h2>
```
```suggestion
    <h2>Securing the VC Data Model</h2>
```

OR13 replied:
This should match the spec title for the TR.

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+    <p>
+      The use of Verifiable Credentials often involves the
+      representation and exchange of structured data in the form of
+      JSON-LD as this is the format of the core data model. While
```
```suggestion
      JSON-LD as this is the structured suffix of the base media types defined in [[VC-DATA-MODEL]]. While
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      representation and exchange of structured data in the form of
+      JSON-LD as this is the format of the core data model. While
+      JSON-LD provides a flexible and extensible format for describing
+      data, it is important to note that it also provides a linkage
```
```suggestion
      RDF, it is important to note that it also provides a linkage
```

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      between the data structure and semantic meaning of data.
+    </p>
+    <p>
+      This section outlines how JSON-LD encoded claimset can be secured
```
```suggestion
      This section outlines how JSON-LD encoded payload can be secured
```

OR13 replied:
one could argue that JSON-LD is not a claimset, since its not JSON... 
... or they could argue that it is a claimset, since it is JSON... 
see https://www.w3.org/TR/json-ld11/#relationship-to-rdf

Based on previous formal objections:

https://www.w3.org/2022/03/did-fo-report.html

I suggest we add an advisement here, and point to https://github.com/w3ctag/design-principles/issues/239

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+      using either JOSE or COSE.
+    </p>
+    <p>
+      A benefit to this approach is that payloads can be made to conform
```
same issue with payload vs claimset... I don't think it makes sense to call a COSE Sign1 "payload" a "claimset"... I think we will need to do a lot of cleanup if we remove the "native JWT encoding from the spec, basically replacing the word "claimset" with "payload" everywhere, and making it content type explicitly JSON-LD ... and NOT... JSON.

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+          <code>cty</code>.
+        </p>
+        <pre class="example vc-jwt" data-vc-kid="#key-0">
+{ "@context": [ "https://www.w3.org/ns/credentials/v2",
```
nit, on formatting this for spec readability

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+        <p>
+          See <a data-cite="rfc7515#section-4.1.10">Common JOSE Header
+          Parameters</a>
+          for additional details regarding usage of <code>typ</code> and
```
`typ` only works for `JSON` payloads... same issue as https://github.com/w3c/vc-jwt/pull/112/files#r1245326420

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+
+        <pre class="example vc-jwt" title="A secured verifiable
+        presentation with a holder identifier">
+{ "@context": [ "https://www.w3.org/ns/credentials/v2",
```
nit on formatting

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+          with COSE.
         </p>
+        <p>[[rfc8152]] MAY be used to secure this media type.</p>
+        <p>When using this approach, the <code>type (TBD)</code> MUST be
```
reminder about this `typ` issue... cc @selfissued

OR13 commented on index.html:
```diff
@@ -1,1046 +1,1078 @@
+        Binary Object Representation (CBOR) Tags</a> for additional
+        details.</p>
+      </section>
+      <p class="issue" data-number="67"></p>
```
nit, some text here would help with future editorial cleanup

mprorock commented:
> I have nits, but I don't know if any of them matter.
> 
> Please add the issue markers for tag, to assist them in their review

yeah - the nits are in sections i didn't touch - only non-format changes were in doc title and abstract/intro
will add issue markers separately in a follow on PR

andresuribe87 reviewed (approved):
> with new short name "vc-jose-cose"
Curious how this will be performed? Will this be moved to a new github repo?

mprorock commented:
@selfissued need to get this one merged so that we are building on consistent formatting - blocking addressing of additional PRs to address notes from @OR13 above

mprorock commented:
since this is hard to read due to reformatting, the actual changes are here (everything else untouched):

changes:
```
<title>Securing Verifiable Credentials using JOSE and COSE</title>
```

```
shortName: "vc-jose-cose",
```

added/adjusted:
```
 <section id="introduction">
    <h2>Introduction</h2>
    <p>
      This specification describes how to secure media types
      expressing Verifiable Credentials and Verifiable Presentations
      as described in the [[VC-DATA-MODEL]], using approaches
      described by the COSE and JOSE working groups at IETF. This
      includes JWTs [[RFC7519]], JWS [[RFC7515]], and COSE [[RFC8152]]
      and provides an approach using well defined content types
      [[RFC6838]] to identify that the type of data being secured
      conforms with the [[VC-DATA-MODEL]].
    </p>
    <p>
      JSON Web Signatures (JWS) [[RFC7515]] provide a standardized
      mechanism for digitally signing JSON data. It provides a means
      to ensure the integrity, authenticity, and non-repudiation of
      the information contained in a JSON object. JWS enables secure
      communication and trust between parties by adding a
      cryptographic signature to a payload. This signature is
      generated using a private key and can be verified using the
      corresponding public key. JWS enables content integrity, authentication and non repudiation, 
      protecting data from unauthorized modifications. The
      versatility and wide adoption of JWS make it an essential
      component for verifying the authenticity and integrity of
      data in various applications, including
      authentication, authorization, and secure data exchange. These
      properties make JWS especially well suited to securing the
      JSON-LD [[VC-DATA-MODEL]] and through this to provide a means of
      supporting the three party model (issuer, verifier, holder) for
      exchange of credentials enabled in verifiable credenials.
    </p>
```

added:
```
<p>
      CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
      specification that defines a framework for signing and
      encrypting CBOR (Concise Binary Object Representation) [RFC8949]
      data structures. COSE provides a standardized way to secure the
      integrity, authenticity, and confidentiality of CBOR-encoded
      information. It offers a flexible and extensible set of
      cryptographic options, allowing for a wide range of algorithms
      to be used for signing and encryption.

      COSE supports two main operations: signing and encryption. For
signing, COSE allows the creation of digital signatures over
      CBOR data using various algorithms such as HMAC, RSA, ECDSA, or
      EdDSA. These signatures provide assurance of data integrity and
      authenticity. COSE also supports encryption, enabling the
      confidentiality of CBOR data by encrypting it with symmetric or
      asymmetric encryption algorithms.
    </p>
  </section>
  ```

selfissued commented on index.html:
```diff
@@ -1,1046 +1,1080 @@
+      CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
+      specification that defines a framework for signing and
+      encrypting CBOR (Concise Binary Object Representation) [RFC8949]
+      data structures. COSE provides a standardized way to secure the
```
CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
      specification that defines a framework for signing and
      encrypting CBOR (Concise Binary Object Representation) [RFC8949]
      data structures.
->
CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
      specification that defines a framework for representing signed and
      encrypted data using (Concise Binary Object Representation) [RFC8949]
      data structures.

selfissued replied:
It's CWT that defines using COSE to sign CBOR claims - not COSE.  If we want to talk about using COSE to secure CBOR data structures then reference CWT [RFC 8392].

selfissued reviewed (approved):
Approved with two comments to address before merging.

mprorock commented on index.html:
```diff
@@ -1,1046 +1,1080 @@
+      </p> 
+    </p>
+    <p>
+      CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
+      specification that defines a framework for signing and
+      encrypting CBOR (Concise Binary Object Representation) [RFC8949]
+      data structures. COSE provides a standardized way to secure the
```
```suggestion
      CBOR Object Signing and Encryption (COSE) [[RFC8152]] is a
      specification that defines a framework for representing signed and
      encrypted data using (Concise Binary Object Representation) [RFC8949]
      data structures. COSE provides a standardized way to secure the
```

OR13 commented:
Editorial, merging

TallTed commented on index.html:
```diff
@@ -1,1046 +1,1080 @@
+    </p>
+    <p>
+      JSON Web Signatures (JWS) [[RFC7515]] provide a standardized
+      mechanism for digitally signing JSON data. It provides a means
+      to ensure the integrity, authenticity, and non-repudiation of
+      the information contained in a JSON object. JWS enables secure
+      communication and trust between parties by adding a
+      cryptographic signature to a payload. This signature is
+      generated using a private key and can be verified using the
+      corresponding public key. JWS enables content integrity, authentication and non repudiation, 
+      protecting data from unauthorized modifications. The
```
Needs re-reading...
```
It provides a means
to ensure the integrity, authenticity, and non-repudiation of
the information contained in a JSON object. 
```
is redundant with
```
JWS enables content integrity, authentication and non repudiation, 
protecting data from unauthorized modifications.
```
